test/docs_orm_html_redirect/test.sh

```diff
@@ -0,0 +1,84 @@
+#!/bin/bash -e
+
+# The content of this generated file can be copy-pasted to
+# relevant places in roles/websites/files/etc/httpd/conf.d/docs.hibernate.org-rewriterules.confpart
+rm -f rewriterules.tmp
+
+rewriterule() {
+  rewritecond="$1"
+  rewriterule_match="$2"
+  rewriterule_replace="$3"
+  perl -wpe "s,$rewriterule_match,$rewriterule_replace,g if m,$rewritecond,"
+  {
+    echo "RewriteCond %{REQUEST_URI} ^$rewritecond"
+    echo "RewriteRule $rewriterule_match $rewriterule_replace $4"
+  } >> rewriterules.tmp
+}
+
+toindex_part() {
+  rewriterule \
+    '^/(?:orm/[^/]+|stable/orm)/' \
+    "^(.*)/($1)$2$" \
+    '$1/$2' \
+    '[R=301,L,E=nocache:1]'
+}
+
+toindex() {
+  toindex_part "dialect/" "dialect.html" \
+    | toindex_part "integrationguide/html_single/" "Hibernate_Integration_Guide.html" \
+    | toindex_part "introduction/html_single/" "Hibernate_Introduction.html" \
+    | toindex_part "migration-guide/" "migration-guide.html" \
+    | toindex_part "querylanguage/html_single/" "Hibernate_Query_Language.html" \
+    | toindex_part "repositories/html_single/" "Hibernate_Data_Repositories.html" \
+    | toindex_part "userguide/html_single/" "Hibernate_User_Guide.html" \
+    | toindex_part "whats-new/" "whats-new.html"
+}
+
+fromindex_part() {
+  rewriterule \
+    '^/(?:orm/[3456]\.[^/]+|orm/7\.[01]|stable/orm|orm/current)/' \
```

Comment:
We'll need to change this (and adapt the corresponding rewrite rules) when 7.2 becomes stable:

```suggestion
    '^/(?:orm/[3456]\.[^/]+|orm/7\.[01])/' \
```